spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java

```diff
@@ -352,25 +352,25 @@ private void applyDeleteMutations(Iterable<?> objects, List<Mutation> mutations)
 	}
 
 	@Override
-	public void delete(Class entityClass, Key key) {
+	public <T> void delete(Class<T> entityClass, Key key) {
 		applyDeleteMutations(entityClass, KeySet.newBuilder().addKey(key).build(),
 				Collections.singletonList(this.mutationFactory.delete(entityClass, key)));
 	}
 
 	@Override
-	public void delete(Class entityClass, KeySet keys) {
+	public <T> void delete(Class<T> entityClass, KeySet keys) {
 		applyDeleteMutations(entityClass, keys, Collections
 				.singletonList(this.mutationFactory.delete(entityClass, keys)));
 	}
 
-	private void applyDeleteMutations(Class entityClass, KeySet keys, List<Mutation> mutations) {
+	private <T> void applyDeleteMutations(Class<T> entityClass, KeySet keys, List<Mutation> mutations) {
 		maybeEmitEvent(new BeforeDeleteEvent(mutations, null, keys, entityClass));
 		applyMutations(mutations);
 		maybeEmitEvent(new AfterDeleteEvent(mutations, null, keys, entityClass));
 	}
 
 	@Override
-	public long count(Class entityClass) {
+	public <T> long count(Class<T> entityClass) {
```

Comment:
T is never used, please remove

Reply:
np, but it was declared in SpannerOperations.
It was done mainly to remove the warning with code inspection about "Raw use of parametrized class 'Class'".
Still, remove?

Reply:
Matching what's in `SpannerOperations` makes sense to me.

Reply:
Makes sense, let's keep it.